pkg/controller/instanceset/instance_util.go

```diff
@@ -236,16 +245,55 @@ func GenerateAllInstanceNames(parentName string, replicas int32, templates []Ins
 	return instanceNameList
 }
 
+func GenerateAllInstanceNamesWithOrdinals(parentName string, replicas int32, templates []InstanceTemplate, offlineInstances []string, defaultTemplateOrdinals workloads.Ordinals) ([]string, error) {
```

Comment:
@weicao The reason I divided this into two functions is that Ordinals are only set on the `InstanceSet` and are used exclusively within the InstanceSet Controller. In fact, `GenerateAllInstanceNamesWithOrdinals` is specifically designed for the InstanceSet Controller. 

If these were merged into one function, it would mean that callers other than the InstanceSet Controller would also need to make modifications and input some unnecessary parameters.

Reply:
And I think that other controllers, such as the Component Controller, actually don't need to concern themselves with Ordinals.